### PR TITLE
Increase pending allowed

### PR DESCRIPTION
**Pull request by enxebre**
Re open https://github.com/openshift/cluster-machine-approver/pull/33 squashed in one commit

### CODE REVIEW

**Comment by enxebre** [comment]
cc @michaelgugino @ingvagabund @mrunalp

File: main.go [hunk 1]
```diff
@@ -111,20 +111,48 @@ func (c *Controller) handleNewCSR(key string) error {
...
+		klog.Errorf("Pending CSRs: %d; Pending Machines %d", pending, pendingMachinesCount)
 		return nil
 	}
 
-	machines, err := c.machines.List(metav1.ListOptions{})
+	parsedCSR, err := parseCSR(csr)
 	if err != nil {
-		return fmt.Errorf("failed to list machines: %v", err)
+		klog.Infof("error parsing request CSR: %v", err)
+		return nil
```
Comment (frobware) [inline main.go]: Why does this no longer return an error?

File: main.go [hunk 2]
```diff
@@ -111,20 +111,48 @@ func (c *Controller) handleNewCSR(key string) error {
...
+	}
+	maxPendingTotal := maxPendingCSRs + pendingMachinesCount + approvedCount
+	// If for some reason we see a really high approved count, just set
+	// absolute max to total machines + maxPendingCSRs
+	if approvedCount > len(machines.Items) {
+		maxPendingTotal = len(machines.Items) + maxPendingCSRs
+	}
+
+	if pending := recentlyPendingCSRs(c.indexer); pending > maxPendingTotal {
+		klog.Error("Difference between pending CSRs and machines > 100", pending)
```
Comment (frobware) [inline main.go]: We should avoid the hardcoded "> 100"; format appropriately.

File: main.go (hunk 1 above)
Comment (enxebre) [reply main.go]: this behaviour is un changed. If failed to parse the queue forget this key so AddRateLimited is not applied for future processing of updates https://github.com/openshift/cluster-machine-approver/blob/master/main.go#L153

File: main.go [hunk 3]
```diff
@@ -111,9 +112,16 @@ func (c *Controller) handleNewCSR(key string) error {
...
+	machines, err := c.machines.List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list machines: %v", err)
+	}
+
+	maxPending := getMaxPending(machines.Items)
+
+	if pending := recentlyPendingCSRs(c.indexer); pending > maxPending {
+		klog.Errorf("Pending CSRs: %d; Max pending allowed: %d", pending, maxPending)
+		return fmt.Errorf("Difference between pending CSRs and machines > 100: %v", pending)
```
Comment (enxebre) [inline main.go]: @michaelgugino we should preserve the return nil here so upcoming CSRs are ignored

**Pull request merged**